### PR TITLE
[RLlib] Optimize the trainer runner test, add method for shutting down a trainer runner and releasing resources

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: avnish <avnish@anyscale.com>

Add a shutdown method to the trainer runner, that way training actors can be gracefully cleaned up. Clean up the trainer runner tests by removing time.sleep statements and separate out distributed and non distributed test cases.

Run local tests from inside of a ray task. This is because tensorflow has a problem with running distributed and non distributed in the same python process. The gpu memory memory ends up being blocked, causing an oom. Running within a ray task allows for the tensorflow resources to be gracefully reaped.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_trainer/tests/test_trainer_runner.py [hunk 1]
```diff
@@ -16,94 +14,84 @@
...
-        # GPU memory footprint of tf.
-        fws = ["torch"]
-        scaling_modes = ["local-cpu", "local-gpu"]
-        test_iterator = itertools.product(fws, scaling_modes)
+class TestTrainerRunner(unittest.TestCase):
+    def setUp(self) -> None:
+        ray.init()
+
+    def tearDown(self) -> None:
+        ray.shutdown()
```
Comment (gjoliver) [inline rllib/core/rl_trainer/tests/test_trainer_runner.py:39]: this is good.

**Review by gjoliver — APPROVED** [review]
assuming there is no real logic changes.

File: rllib/core/rl_trainer/tests/test_trainer_runner.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/core/rl_trainer/tests/test_trainer_runner.py:39]: oh you can totally shutdown ray with each test method. Good idea 👍

File: rllib/core/rl_trainer/trainer_runner.py [hunk 2]
```diff
@@ -255,3 +256,8 @@ def set_state(self, state: List[Mapping[ModuleID, Mapping[str, Any]]]) -> None:
             for worker in self._workers:
                 refs.append(worker.set_state.remote(state))
             ray.get(refs)
+
+    def shutdown(self):
```
Comment (kouroshHakha) [inline rllib/core/rl_trainer/trainer_runner.py:260]: can we modify the `__del__` method to also call shutdown? before deleting the object?

**Review by kouroshHakha — COMMENTED** [review]
Looks good. Just one comment about deleting the runner. We need to make sure if runner is gced the resources are released. so we need to upgrade its del method to shutdown the backend.

File: rllib/core/rl_trainer/trainer_runner.py (hunk 2 above)
Comment (avnishn) [reply rllib/core/rl_trainer/trainer_runner.py:260]: yerp

Comment (avnishn) [reply rllib/core/rl_trainer/trainer_runner.py:260]: done

**Pull request merged**